irohad/main/impl/consensus_init.cpp

```diff
@@ -82,9 +85,12 @@ namespace iroha {
       std::shared_ptr<consensus::yac::Yac> YacInit::createYac(
           ClusterOrdering initial_order,
           const shared_model::crypto::Keypair &keypair,
-          std::chrono::milliseconds delay_milliseconds) {
+          std::chrono::milliseconds delay_milliseconds,
+          std::shared_ptr<
+              iroha::network::AsyncGrpcClient<google::protobuf::Empty>>
+              async_call) {
         return Yac::create(YacVoteStorage(),
-                           createNetwork(),
+                           createNetwork(async_call),
```

Comment:
`shared_ptr` is passed by value, thus copied, so you can `createNetwork(std::move(async_call))`, I assume